### PR TITLE
Throwaway PR for Issue #744 - Support ES aggregations

### PR DESCRIPTION
**Pull request by arunma**

### CODE REVIEW

File: elasticsearch/src/main/scala/akka/stream/alpakka/elasticsearch/ElasticsearchSourceStage.scala [hunk 1]
```diff
@@ -161,6 +167,14 @@ sealed class ElasticsearchSourceLogic[T](indexName: String,
       case ScrollResponse(Some(error), _) =>
         failStage(new IllegalStateException(error))
         false
+      /*
+      TODO Review This has to be mapped for push
+      Also, the isDefined and get looks super ugly. This would be changed once we have a clarity over how to return aggregated results
+       */
+      case ScrollResponse(None, Some(result)) if result.messages.isEmpty && result.aggregate.isDefined =>
```
Comment (ennru) [inline elasticsearch/src/main/scala/akka/stream/alpakka/elasticsearch/ElasticsearchSourceStage.scala]: This part is tricky indeed. Especially as T is variable. I've no good idea.

**Comment by arunma** [comment]
:-) Thanks a lot @ennru  I always knew that the akka community is very welcoming.

File: elasticsearch/src/main/scala/akka/stream/alpakka/elasticsearch/ElasticsearchSourceStage.scala [hunk 2]
```diff
@@ -20,7 +20,13 @@ import scala.collection.JavaConverters._
 final case class OutgoingMessage[T](id: String, source: T, version: Option[Long])
 
 case class ScrollResponse[T](error: Option[String], result: Option[ScrollResult[T]])
-case class ScrollResult[T](scrollId: String, messages: Seq[OutgoingMessage[T]])
+/* TODO Review
+1. Are aggregated results of type T?
```
[the remark this answers is not in the document]
Comment (arunma) [reply elasticsearch/src/main/scala/akka/stream/alpakka/elasticsearch/ElasticsearchSourceStage.scala]: Makes perfect sense.  Also makes the job much simpler.  I'll have the PR updated (while being incomplete).

File: elasticsearch/src/main/scala/akka/stream/alpakka/elasticsearch/ElasticsearchSourceStage.scala [hunk 3]
```diff
@@ -20,7 +20,13 @@ import scala.collection.JavaConverters._
...
+2. What if the user would like to get the Hits (size >0) and Aggregates?
```
[the remark this answers is not in the document]
Comment (arunma) [reply elasticsearch/src/main/scala/akka/stream/alpakka/elasticsearch/ElasticsearchSourceStage.scala]: Absolutely. That would make the design much simpler. Honestly, I've never done a hits and an aggregate query.

File: elasticsearch/src/main/scala/akka/stream/alpakka/elasticsearch/scaladsl/ElasticsearchSource.scala [hunk 4]
```diff
@@ -147,6 +147,8 @@ object ElasticsearchSource {
         case None => {
           val scrollId = jsObj.fields("_scroll_id").asInstanceOf[JsString].value
           val hits = jsObj.fields("hits").asJsObject.fields("hits").asInstanceOf[JsArray]
+          //TODO Review - Two variants here : 1. Global aggregations and 2. Buckets
+          val aggs = Option(jsObj.fields("aggregations").asJsObject)
```
[the remark this answers is not in the document]
Comment (arunma) [reply elasticsearch/src/main/scala/akka/stream/alpakka/elasticsearch/scaladsl/ElasticsearchSource.scala]: Ah. Thanks !! Never knew that trick.

File: elasticsearch/src/main/scala/akka/stream/alpakka/elasticsearch/ElasticsearchSourceStage.scala (hunk 1 above)
Comment (arunma) [reply elasticsearch/src/main/scala/akka/stream/alpakka/elasticsearch/ElasticsearchSourceStage.scala]: Me neither. I'll think about this and bother you if I come up with something saner.

**Comment by ennru** [comment]
Team communication happens on the Gitter akka/dev channel.

We've no agenda for Alpakka, whatever contributors gets considered for inclusion.

Try to add more [self contained examples](https://developer.lightbend.com/docs/alpakka/current/examples/index.html) and you might find things in the APIs or docs that can be improved.

**Comment by ennru** [comment]
If you'd like to do a Elasticsearch example you could make up a scenario where you extract eg. a book descriptions database table with Slick, convert it to JSON and make it searchable from ES.

**Comment by arunma** [comment]
Sounds great @ennru   Just to be clear, if you are to explain the purpose of the self-contained examples, what would that be? Can I say that they are not expected to be dummy examples with mocked up data but should simulate a real world problem/use-case?  Going by your example, it's common that we have a secondary searchable ES index over a database. Or log data made easier for search etc?

**Comment by ennru** [comment]
Yes, the examples are meant to illustrate how an end-to-end integrations with Akka Streams and Alpakka look like. 
In case of my suggestion above it would illustrate how to create that kind of search index in ES for an exisiting database.

**Comment by arunma** [comment]
Thanks a ton @ennru Greatly appreciate your guidance.

**Comment by ennru** [comment]
I'll close this one than, hope to see more things coming from your side soon.

**Comment by arunma** [comment]
@ennru Thanks !! Absolutely !

**Pull request closed without merging**